File: medium/0078_subsets/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<std::vector<int>> subsets(std::vector<int> &nums) {
    std::vector<std::vector<int>> subsets{{}};

    for (size_t i{}; i < nums.size(); ++i) {
      subsets.push_back({nums[i]});

      size_t limit = subsets.size() - 1;
      for (size_t j = 1; j < limit; ++j) {
        std::vector<int> subset{subsets[j]};
        subset.insert(subset.end(), subsets[limit].begin(),
                      subsets[limit].end());
        subsets.push_back(subset);
      }
    }

    return subsets;
  }
};

File: medium/0078_subsets/solution.py
class Solution:
    def subsets(self, nums: list[int]) -> list[list[int]]:
        if len(nums) == 0:
            return [[]]

        if len(nums) == 1:
            return [[], *[[num] for num in nums]]

        subsets_list = [[]]

        for num in nums:
            for i in range(len(subsets_list)):
                subsets_list.append(subsets_list[i] + [num])

        return subsets_list
